### PR TITLE
@sweir => Auction Cleanup 

### PR DESCRIPTION
**Pull request by damassi**
This PR cleans up a few rough edges around Auctions in Force:

**1** - After bidder registration, when clicking into an auction artwork and then hitting back `/confirm-registration` would trigger the previously dismissed registration status modal

**To Test:**

```sh
git checkout -b damassi-auctions-polish master
git pull https://github.com/damassi/force.git auctions-polish
yarn install && yarn start
```

- Log out and create a new account
- Navigate to http://localhost:5000/auction/members-only and click `Register to Bid` 
- Fill out form and submit
- Click the `x` or surrounding area and notice that the `/confirm-registration` route fragment is replaced in the URL bar

**2** - On `/artwork/:id`, an additional client-side query was necessary to update the "Bid' button so that it displays the right state. Now this is performed on the server enabling server to render correct state, and polled on the client should updates be needed. (Note that as this is bootstrap data, existing tests cover button state.)

**To Test:**
- Log out and create a new account
- Navigate to http://localhost:5000/auction/members-only and click `Register to Bid` 
- Fill out form and submit
- Navigate to http://localhost:5000/artwork/cesar-trinca-painting and note that the button state is immediately correct (`Registration Pending`)

**3** - After bidder registration, the status modal is triggered twice with the second appearance occurring due to a strange hard page refresh thats triggered due to discrepancies between form updates made during registration and sync checks in https://github.com/artsy/force/blob/master/desktop/lib/global_client_setup.coffee#L37. See https://artsy.slack.com/archives/C0C4AJ1PF/p1494988497731780 for more background.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -96,7 +96,7 @@
     "qs": "^6.3.0",
     "raven": "^1.2.1",
     "raven-js": "^3.14.2",
-    "rc-slider": "^6.1.2",
+    "rc-slider": "^7.0.8",
```
Comment (damassi) [inline package.json]: Fixes `PropTypes` warnings

Comment (sweir27) [inline package.json]: 🙌

File: desktop/lib/global_client_setup.coffee [hunk 2]
```diff
@@ -34,6 +34,13 @@ ensureFreshUser = (data) ->
   return unless sd.CURRENT_USER
   for attr in ['id', 'type', 'name', 'email', 'phone', 'lab_features',
                'default_profile_id', 'has_partner_access', 'collector_level']
+
+    # NOTE:
```
Comment (sweir27) [inline desktop/lib/global_client_setup.coffee]: Thanks for this!

**Comment by sweir27** [comment]
Looks good to me! Thank you for fixing this annoying issues. 😄

**Pull request merged**